airbyte-integrations/connectors/source-stripe/source_stripe/availability_strategy.py

```diff
@@ -0,0 +1,31 @@
+import logging
+from typing import Optional, Tuple
+
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http.availability_strategy import HttpAvailabilityStrategy
+
+
+class StripeSubStreamAvailabilityStrategy(HttpAvailabilityStrategy):
+    def check_availability(self, stream: Stream, logger: logging.Logger, source: Optional[Source]) -> Tuple[bool, Optional[str]]:
+        """Traverse through all the parents of a given stream and run availability strategy on each of them"""
```

Comment:
This is awesome, exactly what came to mind when thinking about the overlap between availabilitystrategy and substreams. I think it might be something that could be more extendable to other substreams - going to note it down [here](https://github.com/airbytehq/airbyte/issues/21654)

Reply:
I'd recommend a unit test for this, e.g. let's make a parent unavailable and check the availability of the child

Reply:
Agree. Made some unittests for this and made code more optimized